src/components/Message/hooks/useHandleMute.js

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+import { useContext } from 'react';
+import { isUserMuted, validateAndGetMessage } from '../utils';
+import { ChannelContext, TranslationContext } from '../../../context';
+
+/**
+ * @typedef {{
+ *   notify?: import('types').MessageComponentProps['addNotification'],
+ *   getSuccessNotification?: import('types').MessageComponentProps['getMuteUserSuccessNotification'],
+ *   getErrorNotification?: import('types').MessageComponentProps['getMuteUserErrorNotification'],
+ * }} NotificationArg
+ * @type {(message: import('stream-chat').MessageResponse, notification: NotificationArg) => (event: React.MouseEvent<HTMLElement>) => Promise<void>}
+ */
+export const useHandleMute = (message, notifications) => {
+  /**
+   *@type {import('types').ChannelContextValue}
+   */
+  const { client, mutes } = useContext(ChannelContext);
+
+  /**
+   *@type {import('types').TranslationContextValue}
+   */
+  const { t } = useContext(TranslationContext);
+
+  /** @type {(event: React.MouseEvent<HTMLElement>) => Promise<void>} Typescript syntax */
+  return async (event) => {
+    event.preventDefault();
+    const {
+      notify,
+      getSuccessNotification,
+      getErrorNotification,
+    } = notifications;
+
+    if (!t || !message || !notify || !client) {
+      return;
+    }
+    if (!isUserMuted(message, mutes)) {
+      try {
+        if (!message.user) {
```

Comment:
This condition can be put before parent if-else!! And thus can be removed from `else` as well

Reply:
👌 I moved it to the first check.